webview.h

```diff
@@ -745,13 +745,139 @@ inline std::string json_parse(const std::string &s, const std::string &key,
 //
 // ====================================================================
 //
+#include <cstdlib>
+#include <mutex>
+
 #include <JavaScriptCore/JavaScript.h>
 #include <gtk/gtk.h>
 #include <webkit2/webkit2.h>
 
+#ifdef GDK_WINDOWING_X11
+#include <gdk/gdkx.h>
+#endif
+
+#include <fcntl.h>
+#include <sys/stat.h>
+
 namespace webview {
 namespace detail {
 
+// Namespace containing workaround for WebKit 2.42 when using NVIDIA GPU
+// driver.
+// See WebKit bug: https://bugs.webkit.org/show_bug.cgi?id=261874
+// Please remove all of the code in this namespace when it's no longer needed.
+namespace webkit_dmabuf {
+
+// Get mutex used to make sure we don't call getenv/setenv concurrently.
+static std::mutex &get_env_mutex() {
+  static std::mutex mutex;
+  return mutex;
+}
+
+// Get environment variable. Use mutex returned by get_env_mutex().
+static inline std::string get_env(const std::string &name, std::mutex &mutex) {
+  std::lock_guard<std::mutex> lock{mutex};
```

Comment:
Is the mutex really needed? I think we never read the same environment variable that we write to?

Reply:
I wanted to make it clear that these functions couldn't or shouldn't be used concurrently, but I added a comment and removed the locking. Let's keep it simple until the mutex is really needed!